src/core/gui/inputdevices/DeviceId.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Xournal++
+ *
+ * Device identifier
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+struct DeviceId {
+    DeviceId() = default;
+    explicit DeviceId(const void* id): id(id) {}
+    void clear() { id = nullptr; }
+    operator bool() const { return id != nullptr; }
```

Comment:
Conversion operators, should be marked "explicit", especially boolean conversions. 
Also, the equality and inequality operator is missing; the code only compiles, because `lhs != rhs` is implicitly converted to: `(lhs.id != nullptr) != (rhs.id != nullptr)` which will most likely be `false != false`

```suggestion
    constexpr explicit operator bool() const { return id != nullptr; }
    constexpr auto operator==(DeviceId rhs) -> bool { return id == rhs.id; }
    constexpr auto operator!=(DeviceId rhs) -> bool { return id == rhs.id; }
```